service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/kubernetesNamespace/KubernetesNamespaceGuardStep.java

```diff
@@ -0,0 +1,58 @@
+package bio.terra.workspace.service.resource.controlled.cloud.azure.kubernetesNamespace;
+
+import bio.terra.common.exception.ConflictException;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.service.resource.controlled.cloud.azure.KubernetesClientProvider;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys;
+import bio.terra.workspace.service.workspace.model.AzureCloudContext;
+import io.kubernetes.client.openapi.ApiException;
+import java.util.UUID;
+import org.springframework.http.HttpStatus;
+
+public class KubernetesNamespaceGuardStep implements Step {
```

Comment:
Guard doesn't mean much to me. Suggested name-- `EnforceUniqueKubernetesNamespaceStep`.

Reply:
https://maximegel.medium.com/what-are-guard-clauses-and-how-to-use-them-350c8f1b6fd2#:~:text=What%20is%20a%20guard%20clause,deeper%20and%20less%20meaningful%20error.

Reply:
OK, I see that this is a general pattern.